server/src/main/java/org/folio/reservoir/server/IngestWriteStream.java

```diff
@@ -53,14 +49,6 @@ public Future<Void> write(JsonObject globalRecord) {
     }
     return future
         .compose(x -> {
-          if (moduleJsonPath != null) {
-            JsonObject payload = globalRecord.getJsonObject("payload");
-            Collection<String> strings = moduleJsonPath.executeAsCollection(null, payload);
-            Iterator<String> iterator = strings.iterator();
-            if (iterator.hasNext()) {
-              globalRecord.put("localId", iterator.next().trim());
-            }
-          }
           return storage.ingestGlobalRecord(vertx, sourceId, sourceVersion,
```

Comment:
This used to make localIdPath override default localId.

Reply:
Yes, but AFTER default localId was checked. The code on master is broken: it will skip records that had no default localId but had perfectly good localId looked up with localIdPath. Logging is ofcourse wrong too and logs default id even if localIdPath was specified.

Reply:
TBC: this commit fixes skipping and loggin but also changes behavior: when localIdPath is specified it is used without a fallback. Fallback to default is confusing in this case.